src/inmanta/compiler/config.py

```diff
@@ -0,0 +1,34 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+
+from inmanta.config import Option, is_bool
+
+enable_datatrace: Option[bool] = Option(
+    "compiler",
+    "datatrace_enable",
+    False,
+    "Enables the experimental datatrace application on top of the compiler."
+    " The application should help in identifying the cause of compilation errors"
+    " during the development process.",
+    is_bool,
+)
+
+
+def datatrace_enabled() -> bool:
```

Comment:
why wrap it in a method?

Reply:
To add a layer of abstraction. If we would want to change the section or the name of this option that shouldn't require changes everywhere this check is performed.

Reply:
On second thought, this doesn't make any sense. It's already stored in a variable which provides the desired abstraction. Will fix.

Reply:
I've fixed it. I've also moved the check from `inmanta.execute.scheduler` to `inmanta.compiler` which passes it to the `Scheduler` object. There was an import cycle. Also, it makes more sense to check the config in the compiler module.